src/components/common/SafeTokenWidget/__tests__/SafeTokenWidget.test.tsx

```diff
@@ -81,50 +65,21 @@ describe('SafeTokenWidget', () => {
     expect(result.baseElement).toContainHTML('<body><div /></body>')
   })
 
-  it('Should display 0 if balance has no safe token', async () => {
-    jest.spyOn(useBalances, 'default').mockImplementation(() => ({
-      balances: {
-        fiatTotal: '0',
-        items: [],
-      },
-      loading: false,
-      error: undefined,
-    }))
+  it('Should display 0 if Safe has no SAFE token', async () => {
+    ;(useSafeTokenAllocation as jest.Mock).mockImplementation(() => BigNumber.from(0))
 
     const result = render(<SafeTokenWidget />)
     await waitFor(() => expect(result.baseElement).toHaveTextContent('0'))
   })
 
   it('Should display the value formatted correctly', async () => {
```

Comment:
This test should cover the precision nuance.
I.e, without explicitly passing the precision it renders with a wrong number of decimals **and** when explicitly passing the precision, it formats the number according to that

Reply:
I've set [a better mock value](https://github.com/safe-global/web-core/pull/908/commits/55a334fe0f2c1d615777b48873f10f06428a5c34#diff-cfe34aabfe1308977e7ecf7c7128682037147b447afa6badc9035bf46c10f983R76) for the allocation amount, that results in 4 decimals if formatted badly.